pkg/nsx/services/subnet/subnet.go

```diff
@@ -608,9 +608,9 @@ func (service *SubnetService) MapNSXSubnetToSubnetCR(subnetCR *v1alpha1.Subnet,
 	enabled := false
 	subnetCR.Spec.AdvancedConfig.StaticIPAllocation.Enabled = &enabled
 	if nsxSubnet.AdvancedConfig != nil {
-		// Map EnableVLANExtension from NSX Subnet
-		if nsxSubnet.AdvancedConfig.EnableVlanExtension != nil {
-			subnetCR.Spec.AdvancedConfig.EnableVLANExtension = *nsxSubnet.AdvancedConfig.EnableVlanExtension
+		// Map VlanConection from NSX Subnet
+		if nsxSubnet.VlanConnection != nil {
+			subnetCR.Spec.VLANConnection = *nsxSubnet.VlanConnection
```

Comment:
VlanConnection is not inside AdvancedConfig. So it should be out of if nsxSubnet.AdvancedConfig != nil {

Reply:
Good caught, updated, thanks!